scene/main/viewport.cpp

```diff
@@ -418,288 +418,302 @@ void Viewport::_notification(int p_what) {
 				VS::get_singleton()->multimesh_set_visible_instances(contact_3d_debug_multimesh, point_count);
 			}
 
-			if (physics_object_picking && (to_screen_rect == Rect2() || Input::get_singleton()->get_mouse_mode() != Input::MOUSE_MODE_CAPTURED)) {
+			if (!GLOBAL_GET("physics/common/enable_pause_aware_picking")) {
+				_process_picking(false);
+			}
+		} break;
+		case SceneTree::NOTIFICATION_WM_MOUSE_EXIT:
+		case SceneTree::NOTIFICATION_WM_FOCUS_OUT: {
 
-#ifndef _3D_DISABLED
-				Vector2 last_pos(1e20, 1e20);
-				CollisionObject *last_object = NULL;
-				ObjectID last_id = 0;
-#endif
-				PhysicsDirectSpaceState::RayResult result;
-				Physics2DDirectSpaceState *ss2d = Physics2DServer::get_singleton()->space_get_direct_state(find_world_2d()->get_space());
-
-				if (physics_has_last_mousepos) {
-					// if no mouse event exists, create a motion one. This is necessary because objects or camera may have moved.
-					// while this extra event is sent, it is checked if both camera and last object and last ID did not move. If nothing changed, the event is discarded to avoid flooding with unnecessary motion events every frame
-					bool has_mouse_event = false;
-					for (List<Ref<InputEvent> >::Element *E = physics_picking_events.front(); E; E = E->next()) {
-						Ref<InputEventMouse> m = E->get();
-						if (m.is_valid()) {
-							has_mouse_event = true;
-							break;
-						}
-					}
+			_drop_physics_mouseover();
 
-					if (!has_mouse_event) {
-						Ref<InputEventMouseMotion> mm;
-						mm.instance();
-						mm->set_device(InputEvent::DEVICE_ID_INTERNAL);
-						mm->set_global_position(physics_last_mousepos);
-						mm->set_position(physics_last_mousepos);
-						mm->set_alt(physics_last_mouse_state.alt);
-						mm->set_shift(physics_last_mouse_state.shift);
-						mm->set_control(physics_last_mouse_state.control);
-						mm->set_metakey(physics_last_mouse_state.meta);
-						mm->set_button_mask(physics_last_mouse_state.mouse_mask);
-						physics_picking_events.push_back(mm);
-					}
-				}
+			if (gui.mouse_focus) {
+				//if mouse is being pressed, send a release event
+				_drop_mouse_focus();
+			}
+		} break;
+	}
+}
+
+void Viewport::_process_picking(bool p_ignore_paused) {
+
+	if (!is_inside_tree())
+		return;
+	if (!physics_object_picking)
+		return;
+	if (to_screen_rect == Rect2() || Input::get_singleton()->get_mouse_mode() != Input::MOUSE_MODE_CAPTURED)
```

Comment:
This check seems inverted (see line 421 from the old code).

```suggestion
	if (to_screen_rect != Rect2() && Input::get_singleton()->get_mouse_mode() == Input::MOUSE_MODE_CAPTURED)
```

Reply:
You're right. I actually added the right condition in the PR for _master_. To be honest, I knew I was going to mess the condition up somehow when rewriting it. 😅 I'll fix it in the next push.